ark/utils/test_utils.py

```diff
@@ -0,0 +1,168 @@
+
+    filelocs = {}
+
+    mass_map = tuple(enumerate(channel_names, 1))
+
+    for i, fov in enumerate(fov_names):
+        tif_obj = mi.MibiImage(tif_data[i, :, :, :],
+                               mass_map,
+                               **METADATA)
+
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        tiff.write(tiffpath, tif_obj, dtype=dtype)
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+TIFFMAKERS = {
+    'tiff': _create_tifs,
+    'multitiff': _create_multitiff,
+    'mibitiff': _create_mibitiff,
+}
+
+
+def create_paired_xarray_fovs(base_dir, fov_names, channel_names, img_shape=(1024, 1024),
+                              mode='tiff', delimiter=None, sub_dir=None, fills=None, dtype="int8"):
+
+    if not os.path.isdir(base_dir):
+        raise FileNotFoundError(f'{base_dir} is not a directory')
+        return None
+
+    if fov_names is None or fov_names is []:
+        raise ValueError('No fov names were given...')
+        return None
+
+    if channel_names is None or channel_names is []:
+        raise ValueError('No image names were given...')
+        return None
+
+    if not isinstance(fov_names, list):
+        fov_names = [fov_names]
+
+    if not isinstance(channel_names, list):
+        channel_names = [channel_names]
+
+    # verify fills structure -> dict(fov_name : list [num_channels])
+    if fills is not None:
+        if fills.keys() is not fov_names:
+            raise ValueError(f'fills keys, {fills.keys()}, must match fov_names, {fov_names}')
+            return None
+        for fill in fills.values():
+            if len(fill) != len(channel_names):
+                raise ValueError(f'Not all fills have length {len(channel_names)}')
+                return None
+
+    if fills is not None and not isinstance(fills, list):
+        fills = [fills]
+
+    if fills is not None and (len(fills) != len(channel_names)):
+        raise ValueError('channel_names and fills must be the same length')
+        return None
+
+    filelocs, tif_data = TIFFMAKERS[mode](base_dir, fov_names, channel_names, img_shape, sub_dir,
+                                          fills, dtype)
+
+    if delimiter is not None:
+        fov_ids = [fov.split(delimiter)[0] for fov in fov_names]
+
+    data_xr = xr.DataArray(tif_data,
+                           coords=[fov_ids,
+                                   range(img_shape[0]),
+                                   range(img_shape[1]),
+                                   channel_names],
+                           dims=["fovs", "rows", "cols", "channels"])
+
+    return filelocs, data_xr
+
+
+def clear_directory(dirpath):
+    if not os.path.isdir(dirpath):
+        raise FileNotFoundError(f'{dirpath} is not a valid directory')
+        return
+
+    for subpath in os.listdir(dirpath):
+        if os.path.isfile(subpath):
+            os.remove(subpath)
+        if os.path.isdir(subpath):
+            os.rmdir(subpath)
```

Comment:
Might be easier to use shutil.rmtree(dirpath). That way we can make this a one-liner.

Reply:
We should just make a new temp_dir instead. We can have mutliple context managers within a test, or create a separate test

Reply:
added it since I wasn't if it's more efficient to delete the contents of a temp dir in python, or to just create a new `temp_dir`.  Regardless, I'll go with the latter.